qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -48,6 +48,11 @@ def run_cephfs_shell_cmd(self, cmd, mount_x=None, opts=None, stdin=None):
         return mount_x.client_remote.run(args=args, stdout=StringIO(),
                                            stdin=stdin)
 
+    def get_cephfs_shell_cmd_error(self, cmd, mount_x=None, opts=None,
+                                    stdin=None):
+        return self.run_cephfs_shell_cmd(cmd, mount_x, opts, stdin).stderr.\
+            getvalue().strip()
```

Comment:
You need to modify `run_cephfs_shell_cmd` to save stderr:

`mount_x.client_remote.run(args=args, stdout=StringIO(),`

to

`mount_x.client_remote.run(args=args, stdout=StringIO(), stderr=StringIO(),`

Otherwise we see this failure:

````
2019-11-13T13:52:20.985 INFO:tasks.cephfs.test_cephfs_shell:Running command: cephfs-shell -c /etc/ceph/ceph.conf -- snap create snap1 /data_dir
2019-11-13T13:52:20.985 INFO:teuthology.orchestra.run.smithi203:Running:
2019-11-13T13:52:20.986 INFO:teuthology.orchestra.run.smithi203:> cephfs-shell -c /etc/ceph/ceph.conf -- 'snap create snap1 /data_dir'
2019-11-13T13:52:21.208 INFO:teuthology.orchestra.run.smithi203.stderr:ERROR: snapshot 'snap1' already exists
2019-11-13T13:52:21.208 INFO:teuthology.orchestra.run.smithi203.stderr:To enable full traceback, run the following command:  'set debug true'
2019-11-13T13:52:21.232 INFO:tasks.cephfs_test_runner:test_snap (tasks.cephfs.test_cephfs_shell.TestSnapshots) ... ERROR
...
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:======================================================================
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:ERROR: test_snap (tasks.cephfs.test_cephfs_shell.TestSnapshots)
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:----------------------------------------------------------------------
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:Traceback (most recent call last):
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191112.231923/qa/tasks/cephfs/test_cephfs_shell.py", line 334, in test_snap
2019-11-13T13:52:50.152 INFO:tasks.cephfs_test_runner:    o = self.get_cephfs_shell_cmd_error("snap create snap1 /data_dir")
2019-11-13T13:52:50.153 INFO:tasks.cephfs_test_runner:  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20191112.231923/qa/tasks/cephfs/test_cephfs_shell.py", line 53, in get_cephfs_shell_cmd_error
2019-11-13T13:52:50.153 INFO:tasks.cephfs_test_runner:    return self.run_cephfs_shell_cmd(cmd, mount_x, opts, stdin).stderr.\
2019-11-13T13:52:50.153 INFO:tasks.cephfs_test_runner:AttributeError: 'NoneType' object has no attribute 'getvalue'
````

From: /ceph/teuthology-archive/pdonnell-2019-11-13_04:26:42-fs-wip-pdonnell-testing-20191112.231923-distro-basic-smithi/4502318/teuthology.log

Reply:
@batrick fyi - [4545412](http://qa-proxy.ceph.com/teuthology/mchangir-2019-11-26_12:17:58-fs-wip-cephf-shell-add-snapshot-management-distro-basic-smithi/4545412/teuthology.log), [4545413](http://qa-proxy.ceph.com/teuthology/mchangir-2019-11-26_12:17:58-fs-wip-cephf-shell-add-snapshot-management-distro-basic-smithi/4545413/teuthology.log)

Things look okay here ?

Reply:
Odd... resolving this.